dashboard/db/migrate/20210217022321_create_programming_expressions.rb

```diff
@@ -0,0 +1,12 @@
+class CreateProgrammingExpressions < ActiveRecord::Migration[5.2]
+  def change
+    create_table :programming_expressions do |t|
+      t.string :name, null: false
+      t.string :category
+      t.text :properties
+      t.integer :programming_environment_id, null: false
```

Comment:
I would recommend adding an index on this field. this is the best practice for fields which reference other models by id. Incidentally, you can get this index "for free" if you use the `t.references` syntax, but `t.index` is fine too.